server/internal/core/application.go

```diff
@@ -1,24 +1,62 @@
 package core
 
 import (
+	"github.com/jmoiron/sqlx"
+
+	"github.com/oxidrive/oxidrive/server/internal/config"
+	"github.com/oxidrive/oxidrive/server/internal/core/file"
 	"github.com/oxidrive/oxidrive/server/internal/core/instance"
 	"github.com/oxidrive/oxidrive/server/internal/core/user"
+	fileinfra "github.com/oxidrive/oxidrive/server/internal/infrastructure/file"
+	userinfra "github.com/oxidrive/oxidrive/server/internal/infrastructure/user"
 )
 
 type Application struct {
 	instance instance.Service
+	file     file.Service
 }
 
 type ApplicationDependencies struct {
-	Users user.Users
+	Users         user.Users
+	FilesContent  file.FilesContent
+	FilesMetadata file.FilesMetadata
 }
 
 func NewApplication(deps ApplicationDependencies) *Application {
 	return &Application{
 		instance: instance.NewService(deps.Users),
+		file:     file.NewService(deps.FilesContent, deps.FilesMetadata),
 	}
 }
 
 func (app *Application) Instance() *instance.Service {
 	return &app.instance
 }
+
+func (app *Application) File() *file.Service {
+	return &app.file
+}
+
+func SetupDependencies(cfg config.Config, db *sqlx.DB) ApplicationDependencies {
```

Comment:
We cannot depend on real-word concepts and libraries like sqlx in the `core` module!
The dependencies wiring needs to happen outside, in the main (ideally) or infrastructure layers

Reply:
I see your point, but I would keep it outside of the `main` file, if the complexity grows I want to test the function that initializes the dependencies and testing a binary is horrible. We can move it inside the infra layer and use a decorator or something similar for the setup. wdyt?